src/Commands/MissingCommand.php

```diff
@@ -21,41 +21,85 @@ public function handle(): int
-                    return [
-                        str($key)->limit(20)->value(),
-                        (string) $value['count'],
-                        implode("\n",
-                            array_map(
-                                fn ($file) => str($file)->after(base_path()),
-                                $value['files'],
-                            )
-                        ),
-                    ];
-                })->values()->all()
-        );
+        intro('Using driver: ' . $translator->driver::class);
 
-        if ($sync) {
+        $missingCount = count($missing);
 
-            $translator->setTranslations(
-                locale: $locale,
-                values: array_map(fn () => null, $missing)
-            );
+        if ($missingCount === 0) {
+            info('No missing keys found for locale: ' . $locale);
+            return self::SUCCESS;
+        }
 
-            info(count($missing).' missing keys added to the driver.');
+        $this->components->info("{$missingCount} missing keys detected.");
+        $this->newLine();
 
+        if (!$sync) {
+            table(
+                headers: ['Key', 'Total Count', 'Files'],
+                rows: $this->formatMissingKeys($missing)
+            );
+        }
+
+        if ($sync) {
+            $this->syncMissingKeysWithProgress($translator, $locale, $missing);
         }
 
         return self::SUCCESS;
     }
+
+    /**
+     * Format missing keys for display in the table.
+     *
+     * @param array $missing
+     * @return array
+     */
+    private function formatMissingKeys(array $missing): array
+    {
+        return collect($missing)
+            ->map(fn($value, $key) => [
+                str($key)->limit(20)->value(),
+                (string) $value['count'],
+                implode("\n", $this->formatFiles($value['files'])),
+            ])
+            ->values()
+            ->all();
+    }
+
+    /**
+     * Format file paths for display.
+     *
+     * @param array $files
+     * @return array
+     */
+    private function formatFiles(array $files): array
+    {
+        return array_map(
+            fn($file) => str($file)->after(base_path())->value(),
+            $files
+        );
+    }
+
+    /**
+     * Sync missing keys with the translator driver, showing a progress bar.
+     *
+     * @param object $translator
+     * @param string $locale
+     * @param array $missing
+     * @return void
+     */
+    private function syncMissingKeysWithProgress($translator, string $locale, array $missing): void
+    {
+        $progress = progress(label: 'Syncing...', steps: count($missing));
+        $progress->start();
+
+        foreach ($missing as $key => $value) {
+            $translator->setTranslation($locale, $key, null);
+            $progress->advance();
+        }
+
+        $progress->finish();
+        $this->components->info(count($missing) . ' missing keys have been synced.');
```

Comment:
Using progress is not really relevant here because `setTranslations` will add all missing translations in a single write. Doing a for-loop will just make it less performant.

Reply:
I know that, but installing the package in my big project took a lot of time. So, that's why I added a progress bar.

We can show progress bar with the option, like `--without-progress` or `with-progress`.

Reply:
Understood, but the part taking time is `getMissingTranslations` not `setTranslations`.
I should certainly add a `progress` callback to `getMissingTranslations` so it could be displayed it the terminal. I'm gonna look at it this week, thx for the feedback!